caas/kubernetes/provider/application/trust.go

```diff
@@ -11,23 +11,34 @@ import (
 	rbacv1 "k8s.io/api/rbac/v1"
 )
 
-var defaultApplicationNamespaceRules = []rbacv1.PolicyRule{
-	{
-		APIGroups: []string{""},
-		Resources: []string{"pods", "services"},
-		Verbs: []string{
-			"get",
-			"list",
-			"patch",
+func getDefaultApplicationNamespaceRules(namespace string) []rbacv1.PolicyRule {
+	return []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+			Verbs: []string{
+				"get",
+				"list",
+			},
+			ResourceNames: []string{namespace},
 		},
-	},
-	{
-		APIGroups: []string{""},
-		Resources: []string{"pods/exec"},
-		Verbs: []string{
-			"create",
+		{
+			APIGroups: []string{""},
+			Resources: []string{"pods", "services"},
+			Verbs: []string{
```

Comment:
should we add ResourceNames to this and the the pod exec stuff, no reason an application should be able to change any outside itself

Reply:
Agreed.